src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -2436,7 +2436,7 @@ static OrganizationUserStatusType GetPriorActiveOrganizationUserStatusType(Organ
         return status;
     }
 
-    public async Task CreatePendingOrganization(Organization organization, string ownerEmail)
+    public async Task CreatePendingOrganization(Organization organization, string ownerEmail, string creatorUserName, bool salesAssistedTrialStarted)
```

Comment:
Great suggestion! I tried implementing that but I ran into an issue. I couldn't use `_userService.GetUserName(user)` in `OrganizationService` because then it depended on `UserService` and that is a circular dependency.

Reply:
You can get around this by passing `UserService` into this method as well, e.g.

```cs
public async Task CreatePendingOrganization(Organization organization, string ownerEmail, ClaimsPrincipal user, IUserService userService, bool salesAssistedTrialStarted)
```

It's done elsewhere, e.g. `OrganizationService.AcceptUserAsync`. It's not ideal, we shouldn't have circular dependencies, but I still think it's better.

Reply:
That is a great idea!